pandas/core/apply.py

```diff
@@ -678,6 +700,23 @@ def agg_axis(self) -> Index:
 class FrameApply(NDFrameApply):
     obj: DataFrame
 
+    def __init__(
+        self,
+        obj: AggObjType,
+        func: AggFuncType,
+        raw: bool,
+        result_type: str | None,
+        *,
+        by_row: Literal[False, "compat"] = False,
+        args,
+        kwargs,
+    ) -> None:
+        if by_row is not False and by_row != "compat":
+            raise NotImplementedError(f"by_row={by_row} not implemented")
```

Comment:
From [the docs](https://docs.python.org/3/library/exceptions.html#NotImplementedError), I think  `NotImplementedError` signifies the implementation is currently incomplete, and that users can expect this to be supported once we "get around to it". Can this be a ValueError instead.

Reply:
Also - this can be removed if we're checking in the parent class (since they're the same now).

Reply:
I missed this comment somehow. I've changed it now.